### PR TITLE
GEOMESA-1648 Using shared index api code for z3 index

### PR DESCRIPTION
**Pull request by elahrvivaz**
* Z3 now uses table-sharing along with other indices

Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

**Comment by elahrvivaz** [comment]
note: stacks on top of #1339

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/AccumuloFeatureIndex.scala [hunk 1]
```diff
@@ -21,7 +21,7 @@ import org.locationtech.geomesa.accumulo.data._
 import org.locationtech.geomesa.accumulo.index.legacy.attribute.{AttributeIndexV2, AttributeIndexV3}
 import org.locationtech.geomesa.accumulo.index.legacy.id.RecordIndexV1
 import org.locationtech.geomesa.accumulo.index.legacy.z2.{Z2IndexV1, Z2IndexV2}
-import org.locationtech.geomesa.accumulo.index.z3.{Z3Index, Z3IndexV1, Z3IndexV2}
+import org.locationtech.geomesa.accumulo.index.legacy.z3.{Z3IndexV1, Z3IndexV2, Z3IndexV3}
```
Comment (jnh5y) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/AccumuloFeatureIndex.scala]: Is the new Z3IndexV3 in the legacy package?

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/Z3Index.scala [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+// current version - normal table sharing
+case object Z3Index extends AccumuloFeatureIndex with AccumuloIndexAdapter
+    with Z3Index[AccumuloDataStore, AccumuloFeature, Mutation, Range] {
+
+  import org.locationtech.geomesa.utils.geotools.RichSimpleFeatureType.RichSimpleFeatureType
+
+  val Z3IterPriority = 23
+
+  override val version: Int = 4
```
Comment (jnh5y) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/Z3Index.scala]: Ok... I think it see what's going on...  The current version won't be in the legacy package and the old ones will move there.

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/legacy/z3/Z3IndexV1.scala [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import org.apache.accumulo.core.data.Mutation
+import org.locationtech.geomesa.accumulo.data.{AccumuloDataStore, AccumuloFeature, EMPTY_TEXT}
+import org.locationtech.geomesa.accumulo.index.AccumuloFeatureIndex
+import org.locationtech.geomesa.accumulo.index.AccumuloFeatureIndex.{BinColumnFamily, FullColumnFamily}
+import org.locationtech.geomesa.curve.{BinnedTime, Z3SFC}
+import org.locationtech.geomesa.index.utils.SplitArrays
+import org.opengis.feature.simple.SimpleFeatureType
+
+// initial z3 implementation - only supports points
+case object Z3IndexV1 extends AccumuloFeatureIndex with Z3WritableIndex with Z3QueryableIndex {
```
Comment (jnh5y) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/legacy/z3/Z3IndexV1.scala]: Ok, what's the fate of Z3WritableIndex?  Does that live in the legacy package?

Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/legacy/z3/Z3IndexV1.scala]: yep

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/Z3Index.scala (hunk 2 above)
Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/Z3Index.scala]: yeah the current one will always be Z3Index and then we rename the older ones

**Pull request merged**